### PR TITLE
prov/efa: increase default shm_av_size to 256

### PR DESCRIPTION
**Pull request by wzamazon**
Increase default shm_av_size to 256 to match SMR_MAX_PEERS.

### CODE REVIEW

File: prov/efa/src/rdm/rxr_env.c [hunk 1]
```diff
@@ -45,7 +45,7 @@ struct rxr_env rxr_env = {
 	.use_zcpy_rx = 1,
 	.set_cuda_sync_memops = 1,
 	.zcpy_rx_seed = 0,
-	.shm_av_size = 128,
+	.shm_av_size = 256,
```
Comment (wenduwan) [inline prov/efa/src/rdm/rxr_env.c:48]: Clarify - If the user set `shm_av_size` to 256 on a small instance, will that cause any issue, i.e. oversubscription? Do we need to defend against that?

**Comment by shijin-aws** [comment]
what is the motivation for this PR?

File: prov/efa/src/rdm/rxr_env.c (hunk 1 above)
Comment (wzamazon) [reply prov/efa/src/rdm/rxr_env.c:48]: >  Clarify - If the user set shm_av_size to 256 on a small instance, will that cause any issue, i.e. oversubscription? Do we need to defend against that?

It should out. This handle the case that larger number of processes are used. It does not cause more processes to be used.

**Review by wenduwan — APPROVED** [review]

**Comment by shijin-aws** [comment]
> what is the motivation for this PR?

Nvm, I see this `Increase default shm_av_size to 256 to match SMR_MAX_PEERS.`

**Review by wckzhang — APPROVED** [review]

**Pull request merged**